tuna/miopen/driver/base.py

```diff
@@ -27,62 +27,67 @@
 """Module that encapsulates the DB representation of a Driver cmd"""
 
 from typing import List, Union, Dict, Any
+from re import search
 from abc import abstractmethod
 from sqlalchemy.exc import IntegrityError
 from sqlalchemy.orm import Session
 from sqlalchemy.orm import Query
 from sqlalchemy.inspection import inspect
-from tuna.dbBase.sql_alchemy import DbSession
 from tuna.utils.logger import setup_logger
+from tuna.dbBase.sql_alchemy import DbSession
 from tuna.utils.db_utility import build_dict_val_key, get_session_val_map
 from tuna.miopen.db.miopen_tables import TensorTable
 from tuna.miopen.db.miopen_tables import ConvolutionConfig
-from tuna.miopen.utils.metadata import TENSOR_PRECISION
-from tuna.miopen.utils.parsing import parse_line
+from tuna.miopen.utils.metadata import TENSOR_PRECISION, DIR_MAP
+from tuna.miopen.utils.parsing import parse_line, get_fds_from_cmd
+from tuna.driver import DriverBase
 
-LOGGER = setup_logger('driver_base')
+LOGGER = setup_logger('MIOpenDriver_driver_base')
 
 
-class DriverBase():
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=too-many-public-methods
+class MIOpenDriver(DriverBase):
   """Represents db tables based on ConfigType"""
   tensor_attr: List[str] = [column.name for column in inspect(TensorTable).c]
   tensor_id_map: Dict[str, int] = {}
 
-  def __init__(self,
-               line: str = str(),
-               db_obj: ConvolutionConfig = None) -> None:
-    if line:
-      if not self.construct_driver(line):
-        raise ValueError(f"Error creating Driver from line: '{line}'")
-    elif db_obj:
-      if not self.construct_driver_from_db(db_obj):
-        raise ValueError(
-            f"Error creating Driver from db obj: '{db_obj.to_dict()}'")
-    else:
-      raise ValueError(
-          "Error creating Driver. MIOpen Driver cmd line or db_obj required")
+  def __init__(self, line: str = str(), db_obj: ConvolutionConfig = None):
+    super().__init__(line, db_obj)
 
-  def parse_fdb_key(self, line: str):
-    """Overloaded method.Defined in conv&bn driver child class"""
-    raise NotImplementedError("Not implemented")
+  @abstractmethod
+  def compose_tensors(self, keep_id: bool = False) -> dict:
+    """Get tensors needed for DB table based on config type"""
 
   def parse_row(self, db_obj: ConvolutionConfig):
     """Overloaded method.Defined in conv&bn driver child class"""
     raise NotImplementedError("Not implemented")
 
+  @abstractmethod
   def set_cmd(self, data_type: str):
     """Overloaded method.Defined in conv&bn driver child class"""
     raise NotImplementedError("Not implemented")
 
+  @abstractmethod
   def config_set_defaults(self):
     """Overloaded method.Defined in conv&bn driver child class"""
     raise NotImplementedError("Not implemented")
```

Comment:
In the docstrings for these functions, could you please say what they are supposed to do?

Reply:
done!